core/arch/arm/kernel/thread_optee_smc.c

```diff
@@ -523,15 +523,23 @@ static struct mobj *get_rpc_alloc_res(struct optee_msg_arg *arg,
 {
 	struct mobj *mobj = NULL;
 	uint64_t cookie = 0;
+	size_t size = 0;
+	paddr_t pa = 0;
 
 	if (arg->ret || arg->num_params != 1)
 		return NULL;
 
 	if (arg->params[0].attr == OPTEE_MSG_ATTR_TYPE_TMEM_OUTPUT) {
+		pa = arg->params[0].u.tmem.buf_ptr;
+		size = arg->params[0].u.tmem.size;
 		cookie = arg->params[0].u.tmem.shm_ref;
-		mobj = mobj_shm_alloc(arg->params[0].u.tmem.buf_ptr,
-				      arg->params[0].u.tmem.size,
-				      cookie);
+
+		if (core_pbuf_is(CORE_MEM_NSEC_SHM, pa, size))
+			mobj = mobj_shm_alloc(pa, size, cookie);
+#ifdef CFG_CORE_DYN_SHM
```

Comment:
Please fix as @Emantor suggested in https://github.com/OP-TEE/optee_os/pull/4165#discussion_r505638457
It doesn't matter if it's inconsistent with `thread_rpc_alloc_arg()`, this is the preferred style.

Reply:
Maybe a helper function for these 2 occurrences could help, something like `rpc_shm_mobj_alloc(pa, size, cookie)`.